fastedge-rust-sdk/examples/watermark/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+        }
+    };
+
+    // get filename from URL with has format <scheme>://<host>/<filename>
+    let filename = req
+        .uri()
+        .path()
+        .trim_start_matches("/");
+    if filename.is_empty() {
+        return Response::builder()
+            .status(StatusCode::BAD_REQUEST)
+            .body(Body::from("Malformed request - filename expected\n"));
+    }
+
+    // construct S3 signed URL
+    let (signed_url, host) = match sign_s3(filename) {
+        Err(_) => {
+            return Response::builder()
+                .status(StatusCode::INTERNAL_SERVER_ERROR)
+                .body(Body::from("App misconfigured\n"))
+        }
+        Ok((u, h)) => (u, h),
+    };
+
+    /* Actual request to S3 */
+    let s3_req = Request::builder()
+        .method(Method::GET)
+        .uri(signed_url.as_str())
+        .header("Host", host)
+        .body(Body::empty())
+        .unwrap();
+    let rsp = match fastedge::send_request(s3_req) {
+        Err(_) => {
+            return Response::builder()
+                .status(StatusCode::INTERNAL_SERVER_ERROR)
+                .body(Body::empty())
+        }
+        Ok(r) => r,
+    };
+
+    // if response is not 200, just forward it to the caller
+    let (parts, body) = rsp.into_parts();
+    if parts.status != StatusCode::OK {
+        return Ok(Response::from_parts(parts, body))
+        // if you don't want to expose S3 error to the caller, just use
+        // return Response::builder()
+        //     .status(StatusCode::INTERNAL_SERVER_ERROR)
+        //     .body(Body::empty())
+
+    }
+
+    // load response as image
+    let buf = body.to_vec();
+    let img = match load_from_memory(buf.as_slice()) {
+        Ok(i) => i,
+        Err(_e) =>
+            // response body is not a valid image, just return it to the caller without changes
+            return Ok(Response::from_parts(parts, body))
+    };
+
+    // load watermark as image
+    let wm_img = match load_from_memory(wm_buf.as_slice()) {
+        Ok(i) => i,
+        Err(_e) =>
+            // should never happen
+            return Response::builder()
+                .status(StatusCode::INTERNAL_SERVER_ERROR)
+                .body(Body::from("Invalid watermark format\n"))
+    };
+
+    // get opacity from env
+    let opacity = match env::var("OPACITY").ok() {
+        None => DEFAULT_OPACITY,
+        Some(l) => match l.parse::<f32>() {
+            Err(_) => return Response::builder()    // opacity is not a number
+                        .status(StatusCode::INTERNAL_SERVER_ERROR)
+                        .body(Body::from("Invalid opacity value\n")),
+            Ok(v) => {
+                if v < 0.0 || v > 1.0 {     // opacity is not in 0-1.0 range
+                    return Response::builder()
+                        .status(StatusCode::INTERNAL_SERVER_ERROR)
+                        .body(Body::from("Invalid opacity value\n"))
+                }
+                v
```

Comment:
It is more clear to move if to match statement in this way:
```
Ok(v)  if v < 0.0 || v > 1.0  => {     // opacity is not in 0-1.0 range
                    return Response::builder()
                        .status(StatusCode::INTERNAL_SERVER_ERROR)
                        .body(Body::from("Invalid opacity value\n"))
                },
Ok(v) =>  v,
```

Reply:
yep, cleaner this way, changed